internal/controller/argocd.go

```diff
@@ -150,6 +150,31 @@ func (clctrl *ClusterController) InitializeArgoCD() error {
 	return nil
 }
 
+
+func RestartDeployment(ctx context.Context,clientset *kubernetes.Clientset,namespace string,deployment_name string) error {
+
+	deploy,err := clientset.AppsV1().Deployments(namespace).Get(ctx,deployment_name,metav1.GetOptions{})
+	
+	if err != nil {
+		return err
+	}
+
+	if deploy.Spec.Template.ObjectMeta.Annotations == nil {
+        deploy.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+    }
+
+	deploy.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+
+	_,err = clientset.AppsV1().Deployments(namespace).Update(ctx,deploy,metav1.UpdateOptions{})
+	
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
```suggestion
	_,err = clientset.AppsV1().Deployments(namespace).Update(ctx,deploy,metav1.UpdateOptions{})

	if err != nil {
		return fmt.Errorf("unable to update deployment %q: %w", deploy, err)
	}

	return nil
```

I'm down to add some more context to this error just in case the original error doesn't include enough context.